code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -28,10 +29,10 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
-class AEPPushTemplate {
+class AEPPushTemplate implements Serializable {
```

Comment:
i had made these serializable to allow AEPPushTemplate and CarouselPushTemplate objects to be passed via intent extras. i'll revert this and pass the individual values instead.

Reply:
Would setting the common data in a map and passing that via intent be a cleaner solution? I was using the push template like so:
```java
    private static NotificationCompat.Builder createNotificationBuilder(final Context context, final String channelId, final int centerImageIndex, final ArrayList<String> downloadedImageUris, final ArrayList<String> imageCaptions, final ArrayList<String> imageClickActions, final RemoteViews expandedLayout, final RemoteViews smallLayout, final CarouselPushTemplate pushTemplate) {
```
with this function being called by FilmstripCarouselTemplateNotificationBuilder.construct() as well as FilmstripCarouselTemplateNotificationBuilder.handleIntent().

Reply:
i'll just pass the individual values as parameters to `createNotificationBuilder`

Reply:
i've resolved this by setting a static variable containing the CarouselPushTemplate at the time the notification is constructed initially. the push template is guaranteed to not be null as an IllegalArgumentException is thrown if an invalid payload is used.

Reply:
Due to the nature of static  members, they are shared across multiple instances. This will cause problems when multiple notifications are constructed and managed. 
Further, static members cause long retention of classes holding them because we are dealing with Android components like Services and BrodcastReceivers  I strongly recommend passing the indices, uris, deeplinks as arrays of strings or a map.